api/views/CloudRecordSummaryView.py

```diff
@@ -193,15 +193,17 @@ def _paginate_result(self, request, result):
         return serializer.data
 
     def _filter_cursor(self, cursor):
-        """Filter database results based on setting.RETURN_HEADERS."""
-        columns = cursor.description
-        results = []
-        for value in cursor.fetchall():
+        """Filter database results based on settings.RETURN_HEADERS."""
+        results = []  # Return list.
+        for record in cursor.fetchall():  # record refers to one days summary
+            # Construct a new summary with only the values we care about.
             result = {}
-            for index, column in enumerate(value):
-                header = columns[index][0]
-                if header in settings.RETURN_HEADERS:
-                    result.update({header: column})
+            for key, value in record.iteritems():
+                # If the key is int settings.RETURN_HEADERS, add it
```

Comment:
This comment and the next one are mostly obvious and are describing the code. You need to give the reasons why this is being done in this way. What is `RETURN_HEADERS` for? Why are the fields being filtered by that?